### PR TITLE
Start using Google.Api.Gax.Rest for preconditions.

### PR DESCRIPTION
**Pull request by jskeet**
We may want to move CheckEnumValue into GaxRestPreconditions.

### CODE REVIEW

File: src/Google.Storage.V1/Preconditions.cs [hunk 1]
```diff
@@ -23,151 +24,13 @@
...
-        /// <param name="arg1">The second argument to the format string.</param>
-        internal static void CheckArgument<T1, T2>(bool condition, string paramName, string format, T1 arg0, T2 arg1)
-        {
-            if (!condition)
-            {
-                throw new ArgumentException(string.Format(format, arg0, arg1), paramName);
-            }
-        }
-
         internal static T CheckEnumValue<T>(T value, string paramName) where T : struct
```
Comment (chrisdunelm) [inline src/Google.Storage.V1/Preconditions.cs]: Why not put this in GAX too? It'd be nice to get all preconditions out of these individual libraries. I agree this particular method is slightly specialist, but I don't think that matters.

**Comment by chrisdunelm** [comment]
Generally LGTM, just one question.

File: src/Google.Storage.V1/Preconditions.cs (hunk 1 above)
Comment (jskeet) [inline src/Google.Storage.V1/Preconditions.cs]: Yes, I'll do that in a follow-up. I could do it right now, but it's not blocking me as much.
(I couldn't do it while writing this PR, as I was in a hospital and had no signal :)

**Comment by chrisdunelm** [comment]
Just seen that you flagged that up yesterday, sorry - must read learn to read the commit comments!

**Comment by chrisdunelm** [comment]
LGTM

**Pull request merged**